### ISSUE TITLE
[Desktop] Brave Today: Promoted content

### ISSUE THREAD

**Issue by petemill** [brave/brave-browser#13695]
## Description 

Consume promoted content from the Brave Today feed, display with a "Promoted" label, and register views and clicks via the internal ads library.

## Steps to Reproduce

   1. Consume dev feed for Brave Today
   2. Observe there is 1 promoted content item per "page" of Brave Today content. It should be labelled as "Promoted"
   3. Observe that the promoted item image fits and is not cropped like all the other Brave Today content images.

**Comment by LaurenWags** [comment]
Per discussion with @jsecretan , testing of this will be basic checks. I have also marked as `release-notes/exclude` for now. cc @rebron @kjozwiak

**Comment by stephendonner** [comment]
Verification `PASSED`; launched with `--args --enable-logging=stderr --vmodule="*/bat-native-ledger/*"=6,"*/brave_rewards/*"=6,"*/bat-native-ads/*"=6,"*/bat-native-confirmations/*"=6,"*/brave_ads/*"=6,"*/brave_user_model/*"=6 --brave-ads-staging --brave-ads-debug --brave-today-host=brave-today-cdn.brave.software --rewards=staging=true,reconcile-interval=5` using


Brave | 1.21.70 Chromium: 89.0.4389.69 (Official Build) (x86_64)
-- | --
Revision | 6df9174a477a0bea2118cb4669954d517be16104-refs/branch-heads/4389@{#1332}
OS | macOS Version 11.2.2 (Build 20D80)

Verified test plan from https://github.com/brave/brave-browser/issues/13695 and https://github.com/brave/brave-core/pull/7572

Verified on dev feed and staging ads servers on US region

Verified `Promoted content` is shown in the feed:
<img width="1799" alt="Screen Shot 2021-03-01 at 5 20 38 PM" src="https://user-images.githubusercontent.com/387249/109582198-8e979c00-7ab2-11eb-9b8c-04d703dda311.png">


Verified when `Promoted content` is fully in view then user is rewarded for it:
```
[2895:775:0301/171543.799165:VERBOSE3:promoted_content_ad_event_viewed.cc(24)] Viewed promoted content ad with uuid 5ba75daab97c741993961fa0a281e45811dea80699e51a6935f8e28e90af749a and creative instance id 40a69fa9-3814-4ec4-bee8-a176c8fb4690
[2895:775:0301/171543.799453:VERBOSE1:confirmations.cc(90)] Confirming view ad for creative instance id 40a69fa9-3814-4ec4-bee8-a176c8fb4690
[2895:775:0301/171543.800092:VERBOSE1:redeem_unblinded_token.cc(49)] Redeem unblinded token
[2895:775:0301/171543.802108:VERBOSE6:promoted_content_ad_event_viewed.cc(33)] Successfully logged promoted content ad viewed event
[2895:775:0301/171544.139359:VERBOSE1:confirmations.cc(224)] Successfully redeemed unblinded token with confirmation id cc19806a-3f94-4976-b843-d8dbfba1738e, creative instance id 40a69fa9-3814-4ec4-bee8-a176c8fb4690 and view
[2895:775:0301/171544.139784:VERBOSE1:confirmations.cc(245)] Added 1 unblinded payment token with an estimated redemption value of 0.025 BAT, you now have 1 unblinded payment tokens
```

<img width="1792" alt="Screen Shot 2021-03-01 at 5 25 27 PM" src="https://user-images.githubusercontent.com/387249/109584562-a2450180-7ab6-11eb-9bd3-10dbea6b8ddf.png">

Verified the ad is listed in `7 day Ads History`
<img width="1792" alt="Screen Shot 2021-03-01 at 5 25 17 PM" src="https://user-images.githubusercontent.com/387249/109584505-7e81bb80-7ab6-11eb-9ee1-b9390264f1b6.png">


Verified `clicked` event:
```
[2895:775:0301/171552.865848:VERBOSE3:promoted_content_ad_event_clicked.cc(24)] Clicked promoted content ad with uuid 5ba75daab97c741993961fa0a281e45811dea80699e51a6935f8e28e90af749a and creative instance id 40a69fa9-3814-4ec4-bee8-a176c8fb4690
[2895:775:0301/171552.866065:VERBOSE1:confirmations.cc(90)] Confirming click ad for creative instance id 40a69fa9-3814-4ec4-bee8-a176c8fb4690
[2895:775:0301/171552.873557:VERBOSE6:promoted_content_ad_event_clicked.cc(33)] Successfully logged promoted content ad clicked event
[2895:775:0301/171553.036911:VERBOSE1:confirmations.cc(224)] Successfully redeemed unblinded token with confirmation id 0addc02b-6a14-4132-92c8-22ea7f330707, creative instance id 40a69fa9-3814-4ec4-bee8-a176c8fb4690 and click
[2895:775:0301/171553.037791:VERBOSE1:confirmations.cc(245)] Added 1 unblinded payment token with an estimated redemption value of 0 BAT, you now have 2 unblinded payment tokens
```

Confirmed `landed` ad event:

```
[2895:775:0301/171603.795061:VERBOSE1:ad_transfer.cc(140)] Transferred ad for https://newsletter.banklesshq.com/p/nfts-are-mainstream-market-monday
[2895:775:0301/171603.795342:VERBOSE1:confirmations.cc(90)] Confirming landed ad for creative instance id 40a69fa9-3814-4ec4-bee8-a176c8fb4690
[2895:775:0301/171603.796403:VERBOSE1:redeem_unblinded_token.cc(49)] Redeem unblinded token
[2895:775:0301/171603.803468:VERBOSE6:ad_transfer.cc(149)] Successfully logged transferred ad event
[2895:775:0301/171603.977360:VERBOSE1:confirmations.cc(224)] Successfully redeemed unblinded token with confirmation id 697c803e-458d-48aa-b0dc-a5db367e4f9b, creative instance id 40a69fa9-3814-4ec4-bee8-a176c8fb4690 and landed
[2895:775:0301/171603.978069:VERBOSE1:confirmations.cc(245)] Added 1 unblinded payment token with an estimated redemption value of 0 BAT, you now have 3 unblinded payment tokens
```

Verified there is 1 promoted content item per "page" of Brave Today content (see first image)

Verified user can view a notification ad and be rewarded for it:
image

```
[711:775:0301/144404.752855:VERBOSE1:ad_notification_serving.cc(101)] Ad notification delivered:
  uuid: 77a1a548-0236-4ded-8a05-6cc16bd644ee
  creativeInstanceId: ed6c467a-8643-42eb-a7cd-32e878fa3cd4
  creativeSetId: a9f0b49c-3298-4c1b-a8b5-76a71e8d7332
  campaignId: 0e70c893-b0c0-4dab-860d-7e65446f0f0f
  segment: untargeted
  title: This is a test
  body: This is a test
  targetUrl: https://www.brave.com
[711:775:0301/144404.752901:VERBOSE3:ad_notification_event_viewed.cc(24)] Viewed ad notification with uuid 77a1a548-0236-4ded-8a05-6cc16bd644ee and creative instance id ed6c467a-8643-42eb-a7cd-32e878fa3cd4
[711:775:0301/144404.753221:VERBOSE1:confirmations.cc(90)] Confirming view ad for creative instance id ed6c467a-8643-42eb-a7cd-32e878fa3cd4
[711:775:0301/144404.789043:VERBOSE6:ad_notification_event_viewed.cc(33)] Successfully logged ad notification viewed event
```

```
[711:775:0301/144405.131206:VERBOSE1:confirmations.cc(224)] Successfully redeemed unblinded token with confirmation id e0da91c4-c1f0-42ba-bc1a-93a4282098f7, creative instance id ed6c467a-8643-42eb-a7cd-32e878fa3cd4 and view
[711:775:0301/144405.131919:VERBOSE1:confirmations.cc(245)] Added 1 unblinded payment token with an estimated redemption value of 0.025 BAT, you now have 1 unblinded payment tokens
```

Verified user can view a NTP SI ad and be rewarded for it:
<img width="1682" alt="Screen Shot 2021-03-02 at 8 53 43 AM" src="https://user-images.githubusercontent.com/387249/109684130-d3641700-7b34-11eb-8363-e7d03dfc5968.png">

```
[5048:775:0302/085224.446189:VERBOSE3:new_tab_page_ad_event_viewed.cc(24)] Viewed new tab page ad with uuid baebc8a2-6896-49b2-b5b8-c07a0be6c508 and creative instance id 500f0553-b488-4ad9-aead-ff1e8ea87543
[5048:775:0302/085224.446346:VERBOSE1:confirmations.cc(90)] Confirming view ad for creative instance id 500f0553-b488-4ad9-aead-ff1e8ea87543
[5048:775:0302/085224.447322:VERBOSE1:redeem_unblinded_token.cc(49)] Redeem unblinded token
[5048:775:0302/085224.607587:VERBOSE1:confirmations.cc(224)] Successfully redeemed unblinded token with confirmation id b4d5fded-2811-4a8d-9805-7ff1482181e8, creative instance id 500f0553-b488-4ad9-aead-ff1e8ea87543 and view
[5048:775:0302/085224.608144:VERBOSE1:confirmations.cc(245)] Added 1 unblinded payment token with an estimated redemption value of 0.01 BAT, you now have 1 unblinded payment tokens
```
